cpp/src/parquet/arrow/writer.cc

```diff
@@ -526,9 +526,17 @@ Status FileWriter::Make(::arrow::MemoryPool* pool,
 Status GetSchemaMetadata(const ::arrow::Schema& schema, ::arrow::MemoryPool* pool,
                          const ArrowWriterProperties& properties,
                          std::shared_ptr<const KeyValueMetadata>* out) {
-  if (!properties.store_schema()) {
+  if (!properties.store_schema() &&
+      !properties.geo_crs_context()->HasProjjsonCrsFields()) {
```

Comment:
Can we enforce `store_schema` to be true (or throw when it is false) when `HasProjjsonCrsFields()` is true? This is easier to maintain the code.